### PR TITLE
Modify the Stat endpoint to also return the number of failed conduit pods

### PR DESCRIPTION
**Pull request by rmars**
In the ServiceMesh page, I would like to have some indicator of the health of the conduit deployment. This PR adds the ability to count the number of Failed conduit pods to the stat response.

This would add the field `uint64 errored_conduit_pod_count = 6;` in the `PodGroup.Row`

The resource response would now look like:
```
{
  resource: {
    namespace: "",
    type: "namespaces",
    name: "foobar1"
  },
  timeWindow: "1m",
  meshedPodCount: "4",
  totalPodCount: "4",
  erroredConduitPodCount: "0",
  stats: null
}
```

### CODE REVIEW

File: proto/public/api.proto [hunk 1]
```diff
@@ -175,6 +175,8 @@ message StatTable {
 
       uint64 meshed_pod_count = 3;
       uint64 total_pod_count = 4;
+      uint64 errored_conduit_pod_count = 6;
```
Comment (olix0r) [inline proto/public/api.proto]: Can you add a comment here for this new field indicating how these are counted?  Particularly, i'm curious if we are sure these are conduit-specific errors or just any error running a pod?

Comment (rmars) [reply proto/public/api.proto]: added! it's any error such that the pod is `PodFailed`.  the pod has to be a conduit controller pod to be counted.

this field is only useful if we're querying resources that have the conduit controller pods in them (e.g. querying all namespaces or querying namespace conduit from the ServiceMesh page)

Comment (rmars) [reply proto/public/api.proto]: Yeah, that's probably better. The use case is maybe a little to specific as is.

File: proto/public/api.proto [hunk 2]
```diff
@@ -173,8 +173,13 @@ message StatTable {
       Resource resource = 1;
       string time_window = 2;
 
+      // number of pods in this resource that have conduit injected
       uint64 meshed_pod_count = 3;
+      // number of pods in this resource
       uint64 total_pod_count = 4;
+      // number of pods in this resource that have Phase PodFailed
+      uint64 errored_pod_count = 6;
```
Comment (siggy) [inline proto/public/api.proto]: nitpick on naming, as this is intended to communicate the number of pods in the `PodFailed` state, how about just calling it `failed_pod_count`?

File: controller/api/public/stat_summary.go [hunk 3]
```diff
@@ -328,18 +330,22 @@ func metricToKey(metric model.Metric, groupBy model.LabelNames) string {
...
+	meshCount := &podCount{}
 	for _, pod := range pods {
 		meshCount.total++
 		if isInMesh(pod) {
 			meshCount.inMesh++
 		}
+
+		if pod.Status.Phase == apiv1.PodFailed {
+			meshCount.errored++
+		}
```
Comment (siggy) [inline controller/api/public/stat_summary.go]: i'm a little apprehensive about counting failed pods in the `total` and `inMesh` fields. specifically, when i see output like this:
```
$ bin/conduit stat -n emojivoto deploy
NAME       MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99
emoji         1/1     0.00%   0.0rps           0ms           0ms           0ms
vote-bot      1/1         -        -             -             -             -
voting        1/1     0.00%   0.0rps           0ms           0ms           0ms
web           1/1     0.00%   1.0rps           6ms          15ms          19ms
```

i expect those `MESHED` `1/1` values to indicate currently running pods (kubectl requires a `--show-all` flag to show non-running pods). if i were to see `5/6`, it's kind of ambiguous what's running/meshed/failed. i think it's fine to call out failed pods as a separate stat, but perhaps don't count them as meshed or part of total?

File: proto/public/api.proto (hunk 2 above)
Comment (rmars) [reply proto/public/api.proto]: yep, that's a good idea!

File: controller/api/public/stat_summary.go (hunk 3 above)
Comment (rmars) [reply controller/api/public/stat_summary.go]: yeah that makes sense. I can subtract the failed pods from the total count.

File: proto/public/api.proto [hunk 4]
```diff
@@ -173,8 +173,13 @@ message StatTable {
       Resource resource = 1;
       string time_window = 2;
 
+      // number of pods in this resource that have conduit injected
       uint64 meshed_pod_count = 3;
+      // number of pods in this resource
       uint64 total_pod_count = 4;
+      // number of pods in this resource that have Phase PodFailed
```
[the remark this answers is not in the document]
Comment (rmars) [reply proto/public/api.proto]: changed. also changed the comment to reflect that the running_pod_count is the count of running or pending pods

**Pull request merged**